lib/_http_client.js

```diff
@@ -639,7 +639,9 @@ ClientRequest.prototype.setTimeout = function(msecs, callback) {
   }
 
   this.once('socket', function(sock) {
-    sock.setTimeout(msecs, emitTimeout);
+    sock.once('connect', function() {
+      sock.setTimeout(msecs, emitTimeout);
```

Comment:
nit: This is probably over-micro-optimizing, but using `this` instead of `sock` inside the callback would avoid allocating a closure every time here.